lib/ansible/modules/cloud/docker/docker_compose.py

```diff
@@ -717,6 +717,29 @@ def cmd_up(self):
             result['changed'] = build_output['changed']
             result['actions'] += build_output['actions']
 
+        if self.remove_orphans:
+            # containers = list(filter(None, [
+            #     Container.from_ps(self.project.client, container)
+            #     for container in self.project.client.containers(
+            #         all=False,
+            #         filters={'label': ['{0}={1}'.format(LABEL_PROJECT, self.project.name), '{0}={1}'.format(LABEL_ONE_OFF, "False")]})])
+            # )
+
+            containers = self.client.containers(
+                filters={'label': ['{0}={1}'.format(LABEL_PROJECT, self.project.name), '{0}={1}'.format(LABEL_ONE_OFF, "False")]}
+            )
+
+            def _findOrphans():
+                for container in containers:
+                    # service_name = ctnr.labels.get(LABEL_SERVICE)
+                    service_name = container['Labels'][LABEL_SERVICE]
+                    if service_name not in self.project.service_names:
+                        yield container
+
+            orphans = list(_findOrphans())
+            if orphans:
+                result['changed'] = True
```

Comment:
Same here (as in #54317): it doesn't actually remove the orphans, only detects whether there are any.

Reply:
This is the intended behavior since it will cause `self.project.up()` to run in addition to causing the task to report that it's changed.